src/hotspot/share/gc/g1/g1YoungGCPostEvacuateTasks.hpp

```diff
@@ -57,6 +57,7 @@ class G1PostEvacuateCollectionSetCleanupTask1 : public G1BatchedTask {
 // - Purge Code Roots (s)
 // - Reset Hot Card Cache (s)
 // - Update Derived Pointers (s)
+// - Verify Retained Regions
```

Comment:
should add an `(on evacuation failure)` comment, and probably moved close to the comment on "Restore Preserved Marks"